lib/fakes3/server.rb

```diff
@@ -460,7 +462,7 @@ def normalize_request(webrick_req)
       s_req.path = webrick_req.path
       s_req.is_path_style = true
 
-      if !@root_hostnames.include?(host)
+      if !@root_hostnames.include?(host) && !(IPAddr.new(host) rescue nil)
```

Comment:
Ok, that's an alternative solution to fix this. Can we merge this one first without re-testing everything else?